fastlane/Fastfile

```diff
@@ -677,6 +686,41 @@ platform :ios do
     )
   end
 
+  #####################################################################################
+  # upload_gutenberg_sourcemaps
+  # -----------------------------------------------------------------------------------
+  # This lane uploads the Gutenberg source maps
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane upload_gutenberg_sourcemaps [internal:<internal>]
+  #
+  # Example:
+  # bundle exec fastlane upload_gutenberg_sourcemaps internal:true
+  #####################################################################################
+  lane :upload_gutenberg_sourcemaps do | options |
+    build_version = ios_get_build_version(internal: options[:internal])
+    
+    gutenberg_bundle = File.join(PROJECT_ROOT_FOLDER, "Pods/Gutenberg/bundle/ios")
+    sourcemaps_folder = File.join(gutenberg_bundle, "sourcemaps")
+    FileUtils.mkdir_p(sourcemaps_folder)
+    # Name of bundle file must be set to main.jsbundle
+    FileUtils.cp(File.join(gutenberg_bundle, 'App.js'), File.join(sourcemaps_folder, 'main.jsbundle'))
+    FileUtils.cp(File.join(gutenberg_bundle, 'App.js.map'), File.join(sourcemaps_folder, 'main.jsbundle.map'))
+
+    sentry_upload_sourcemap(
+      auth_token: get_required_env("SENTRY_AUTH_TOKEN"),
+      org_slug: 'a8c',
+      project_slug: 'wordpress-ios',
+      version: build_version,
+      dist: build_version,
```

Comment:
Just to check, what's the difference between these two parameters and why is it okay for them to have the same value?

Reply:
They are documented [here](https://docs.sentry.io/platforms/react-native/manual-setup/sourcemaps/#configure-cli), the `version` value should match the one set [here](https://github.com/Automattic/Automattic-Tracks-iOS/blob/aa7040e504377a0322dd91ff366f115e9873a369/Automattic-Tracks-iOS/Crash%20Logging/CrashLogging.swift#L39). However, I'm not sure about what to use for `dist`, when I checked Sentry events, I found that they had the same value (see attached examples) so I went ahead and use the same version value.

<img width="1003" alt="Screenshot 2021-07-01 at 12 48 22" src="https://user-images.githubusercontent.com/14905380/124114781-4642b780-da6d-11eb-8c01-0d8259f7d4cf.png">

I've just checked from where the Sentry iOS SDK takes the `dist` value ([code reference](https://github.com/getsentry/sentry-cocoa/blob/b750f8305717eeb46a11de89b15e2adddc8acc8e/Sources/Sentry/SentryClient.m#L367)) and it's `CFBundleVersion`, so looks like that in the end it's actually the same value 😄 . One thing I could try is to remove the `dist` parameter from the action and check if it affects somehow 👍.

Reply:
I've done a test by removing the `dist` parameter and so far it works, however, this change also implies initializing the SDK in the RN side without that parameter ([code reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/ca6d40ffcd02a18bec7aab258d76c4147ff6e2a3/src/sentry/index.js#L36)).

To be honest, I don't have a strong opinion on whether include it or not, wdyt?

Reply:
@fluiddot I dug a bit into the documents and found a definition of what the `dist` value is meant to be [here](https://docs.sentry.io/platforms/react-native/configuration/releases/) which matches your finding:

> You will also need to pass a distribution to `dist` for events to be attributed correctly to a release. Usually this value is the build number, for example 52. This value needs to be unique to only each release.

Given that, your original approach of passing the build number was spot on.

The snippet in that pages shows the release as something that looks more like a node module name that the build number

```js
Sentry.init({
  release: "my-project-name@2.3.12",
  dist: "52",
});
```

I guess they're making a bit of confusion on their end between `release`, `dist`, `version`, and `build` in the attempt of supporting different platforms? 🤷‍♂️ 

From what I see in your explanation and screenshots, this setup seems fine. 👍